### PR TITLE
Update swagger spec for plugin upgrade

### PR DESCRIPTION
**Pull request by cpuguy83**
This was added in 03c694973968f63743ed53cef83d0b7455695081 but spec was
not updated.

### CODE REVIEW

**Comment by cpuguy83** [comment]
ping @tiborvass @anusha-ragunathan

**Comment by anusha-ragunathan** [comment]
LGTM

File: api/swagger.yaml [hunk 1]
```diff
@@ -6648,6 +6648,69 @@ paths:
...
           type: "string"
       tags: ["Plugin"]
+  /plugins/{name}/upgrade:
+    post:
+      summary: "Upgrade a plugin"
+      operationId: "PluginUpgrade"
+      responses:
+        204:
+          description: "no error"
+        500:
```
Comment (anusha-ragunathan) [inline api/swagger.yaml]: Actually, lets go ahead and add 404 here.

Comment (cpuguy83) [inline api/swagger.yaml]: oh yep!

**Comment by vdemeester** [comment]
@cpuguy83 some validation errors :sweat_smile: 

```
21:09:16 The swagger spec at "api/swagger.yaml" is invalid against swagger specification 2.0. see errors :
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - paths./plugins/{name}/upgrade.post.parameters.in in body should be one of [header]
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - must validate one and only one schema (oneOf)
21:09:16 - paths./plugins/{name}/upgrade.post.parameters.in in body should be one of [header]
```

**Comment by cpuguy83** [comment]
Ok, validations passing now.

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Pull request merged**